### PR TITLE
feat(a11y): Change behavior of some file upload fields

### PR DESCRIPTION
**Pull request by jsit**
Several file upload fields were hiding the main `<input type="file">` and relying on just the label for clicking.

This made these fields inaccessible to tabbing, and possibly some other negative effects of hiding native elements. `<label>` can have a tabindex, but evidently that's not a good practice.

In a couple places here I've just added the `form-control` class to the inputs, which Bootstrap styles nicely.

The places I didn't touch this are the emoji upload page, where there isn't much space, and the Markdown toolbar, where there isn't any room. (I added an `eslint-disable` to ignore the `tabindex` being on the `label` for keyboard access for those.)

I cleaned up the Create Post page to make image uploading a little clearer, too.

Another thing I noticed is that in some cases these upload fields were a `<form>` inside of a `<form>`, which they don't have to be.

Affected pages are:

1. Create Post
2. Personal Settings
3. Create Community
4. Edit Site

### Before

Screenshot 2023-06-25 at 2 33 03 AM

### After

Screenshot 2023-06-26 at 6 38 48 PM

### Before

Screenshot 2023-06-25 at 2 32 24 AM

### After

Screenshot 2023-06-26 at 6 38 43 PM

### Before

Screenshot 2023-06-25 at 2 32 06 AM

### After

Screenshot 2023-06-26 at 6 38 55 PM

### Before

Screenshot 2023-06-25 at 2 31 47 AM

### After

Screenshot 2023-06-26 at 6 39 00 PM

### After

Screenshot 2023-06-25 at 2 25 52 AM

---

# Avatar aspect ratios

This PR also fixes avatar aspect ratios

## Before

Screenshot 2023-07-02 at 5 26 45 PM

## After

Screenshot 2023-07-02 at 5 21 38 PM

Screenshot 2023-07-02 at 5 31 09 PM

### CODE REVIEW

File: src/assets/css/main.css [hunk 1]
```diff
@@ -88,6 +88,15 @@
   color: var(--bs-gray) !important;
 }
 
+input[type="file"]::file-selector-button {
```
Comment (dessalines) [inline src/assets/css/main.css:91]: Does bootstrap 5 not have a a11y-compatible way to file upload buttons? Not sure why this custom css is necessary.

Comment (SleeplessOne1917) [reply src/assets/css/main.css:91]: There is this: https://getbootstrap.com/docs/5.2/forms/input-group/#custom-file-input

Comment (jsit) [reply src/assets/css/main.css]: Ah, of course! Thank you. I've just updated this PR to use those classes.

**Review by SleeplessOne1917 — CHANGES_REQUESTED** [review]
Better than what it used to be, but it could use a few more touches before shipping.

For the site settings, could use a bit of margin between the images and the file inputs. I'm also not a fan of the tiny "x" on the image to remove the image.
image

For create post and create community, a bit of margin between the uploader and image would be good:
image

For edit community and user settings, same critiques as the site settings page apply:
image
image

**Comment by alectrocute** [comment]
I don’t know if it’s related, but the avatar/icons that have a weird aspect ratio (not a circle) has been hurting me for awhile. Could there be an opportunity to fix that here? 😌

**Comment by jsit** [comment]
@alectrocute 

> I don’t know if it’s related, but the avatar/icons that have a weird aspect ratio (not a circle) has been hurting me for awhile. Could there be an opportunity to fix that here? 😌

10,000,000%

**Comment by jsit** [comment]
Thanks for battle-testing this @SleeplessOne1917, I probably won't get to this today.

**Review by SleeplessOne1917 — APPROVED** [review]

**Comment by jsit** [comment]
@SleeplessOne1917 I'm gonna work on those spacing issues before merging this

**Comment by jsit** [comment]
@SleeplessOne1917 Fixed spacing between image and file input:

Screenshot 2023-07-02 at 5 14 21 PM

Screenshot 2023-07-02 at 5 31 51 PM


Screenshot 2023-07-02 at 5 14 17 PM

**Pull request merged**